server/pairing/payload_mounter.go

```diff
@@ -0,0 +1,339 @@
+type AccountPayloadLoader struct {
+	*AccountPayload
+
+	multiaccountsDB *multiaccounts.Database
+	keystorePath    string
+	keyUID          string
+}
+
+func NewAccountPayloadLoader(p *AccountPayload, config *SenderConfig) (*AccountPayloadLoader, error) {
+	ppr := &AccountPayloadLoader{
+		AccountPayload: p,
+	}
+
+	if config == nil {
+		return ppr, nil
+	}
+
+	ppr.multiaccountsDB = config.DB
+	ppr.keyUID = config.KeyUID
+	ppr.password = config.Password
+	ppr.keystorePath = config.KeystorePath
+	return ppr, nil
+}
+
+func (apl *AccountPayloadLoader) LoadFromSource() error {
+	err := apl.loadKeys(apl.keystorePath)
+	if err != nil {
+		return err
+	}
+
+	err = validateKeys(apl.keys, apl.password)
+	if err != nil {
+		return err
+	}
+
+	apl.multiaccount, err = apl.multiaccountsDB.GetAccount(apl.keyUID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (apl *AccountPayloadLoader) loadKeys(keyStorePath string) error {
+	apl.keys = make(map[string][]byte)
+
+	fileWalker := func(path string, fileInfo os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if fileInfo.IsDir() || filepath.Dir(path) != keyStorePath {
+			return nil
+		}
+
+		rawKeyFile, err := ioutil.ReadFile(path)
+		if err != nil {
+			return fmt.Errorf("invalid account key file: %v", err)
+		}
+
+		accountKey := new(keystore.EncryptedKeyJSONV3)
+		if err := json.Unmarshal(rawKeyFile, &accountKey); err != nil {
+			return fmt.Errorf("failed to read key file: %s", err)
+		}
+
+		if len(accountKey.Address) != 40 {
+			return fmt.Errorf("account key address has invalid length '%s'", accountKey.Address)
+		}
+
+		apl.keys[fileInfo.Name()] = rawKeyFile
+
+		return nil
+	}
+
+	err := filepath.Walk(keyStorePath, fileWalker)
+	if err != nil {
+		return fmt.Errorf("cannot traverse key store folder: %v", err)
+	}
+
+	return nil
+}
+
+/*
+|--------------------------------------------------------------------------
+| RawMessagePayload
+|--------------------------------------------------------------------------
+|
+| RawMessagePayloadMounter and RawMessageLoader
+|
+*/
+
+type RawMessagePayloadMounter struct {
+	logger *zap.Logger
+
+	encryptor *PayloadEncryptor
+	loader    *RawMessageLoader
+}
+
+func NewRawMessagePayloadMounter(logger *zap.Logger, pe *PayloadEncryptor, backend *api.GethStatusBackend, config *SenderConfig) *RawMessagePayloadMounter {
+	l := logger.Named("RawMessagePayloadManager")
+	return &RawMessagePayloadMounter{
+		logger:    l,
+		encryptor: pe.Renew(),
+		loader:    NewRawMessageLoader(backend, config),
+	}
+}
+
+func (r *RawMessagePayloadMounter) Mount() error {
+	err := r.loader.LoadFromSource()
+	if err != nil {
+		return err
+	}
+	return r.encryptor.encrypt(r.loader.payload)
+}
+
+func (r *RawMessagePayloadMounter) ToSend() []byte {
+	return r.encryptor.getEncrypted()
+}
+
+func (r *RawMessagePayloadMounter) LockPayload() {
+	r.encryptor.lockPayload()
+}
+
+func (r *RawMessagePayloadMounter) ResetPayload() {
```

Comment:
Super light suggestion, we could use the nice naming `LockPayload` + `UnlockPayload` instead of `LockPayload` + `ResetPayload`. Or change `LockPayload` to `SetupPayload` + `ResetPayload`.

Reply:
Oh, I linked these pairs of actions way too much. It's much clearer now to me.